examples/auction/src/main.leo

```diff
@@ -0,0 +1,23 @@
+record Auction {
```

Comment:
The Leo code here looks great.
I feel like the example could be improved with some comments on how to construct `record` types correctly in Leo.
See https://github.com/AleoHQ/leo/blob/testnet3/examples/message/src/main.leo
for an example.

Reply:
Yes, I've added that, along with additional explanations. I'll push soon (just rebuilding `leo` now, and making sure this example still works on my machine).